hr_expense_invoice/models/hr_expense_expense.py

```diff
@@ -0,0 +1,103 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, fields, api, exceptions, _
+
+
+class HrExpenseExpense(models.Model):
+    _inherit = 'hr.expense.expense'
+
+    @api.model
+    def line_get_convert(self, x, part, date):
+        res = super(HrExpenseExpense, self).line_get_convert(x, part, date)
+        if x.get('invoice'):
+            res['partner_id'] = (
+                x['invoice'].partner_id.commercial_partner_id.id)
```

Comment:
Can you rename your args to something more readable. I know that in hr_expense, the editor uses x and part as parameter, but we the editor's programming standards are quite low. I would suggest line_get_convert(self, line, partner, date).